TLM/TLM/Util/ModsCompatibilityChecker.cs

```diff
@@ -1,82 +1,201 @@
+            {
+                IncompatibleModsPanel panel = UIView.GetAView().AddUIComponent(typeof(IncompatibleModsPanel)) as IncompatibleModsPanel;
+                panel.IncompatibleMods = detected;
+                panel.Initialize();
+                UIView.PushModal(panel);
+                UIView.SetFocus(panel);
             }
+        }
+
+        /// <summary>
+        /// Iterates installed mods looking for known incompatibilities.
+        /// </summary>
+        /// 
+        /// <returns>A list of detected incompatible mods.</returns>
+        /// 
+        /// <exception cref="ArgumentException">Invalid folder path (contains invalid characters, is empty, or contains only white spaces).</exception>
+        /// <exception cref="PathTooLongException">Path is too long (longer than the system-defined maximum length).</exception>
+        public Dictionary<PluginInfo, string> ScanForIncompatibleMods()
+        {
+            Log.Info("Scanning for incompatible mods");
+
+            // list of installed incompatible mods
+            Dictionary<PluginInfo, string> results = new Dictionary<PluginInfo, string>();
+
+            // only check enabled mods?
+            bool filterToEnabled = State.GlobalConfig.Instance.Main.IgnoreDisabledMods;
+
+#if !DEBUG
+            bool offline = IsOffline();
+#endif
 
-            if (incompatibleMods.Count > 0) {
-                Log.Warning("Incompatible mods detected! Count: " + incompatibleMods.Count);
+            // iterate plugins
+            foreach (PluginInfo mod in Singleton<PluginManager>.instance.GetPluginsInfo())
+            {
+                if (!mod.isBuiltin && !mod.isCameraScript && (!filterToEnabled || mod.isEnabled))
+                {
+                    string modName = GetModName(mod);
 
-                if (State.GlobalConfig.Instance.Main.ScanForKnownIncompatibleModsAtStartup) {
-                    IncompatibleModsPanel panel = UIView.GetAView().AddUIComponent(typeof(IncompatibleModsPanel)) as IncompatibleModsPanel;
-                    panel.IncompatibleMods = incompatibleMods;
-                    panel.Initialize();
-                    UIView.PushModal(panel);
-                    UIView.SetFocus(panel);
+                    if (incompatibleMods.ContainsKey(mod.publishedFileID.AsUInt64))
+                    {
+                        Log.Info($"Incompatible mod: {mod.publishedFileID.AsUInt64} - {modName}");
+                        results.Add(mod, modName);
+                    }
+#if !DEBUG
+                    // Workshop TM:PE builds treat local builds as incompatible
+                    else if (!offline && mod.publishedFileID.AsUInt64 == LOCAL_MOD && (modName.Contains("TM:PE") || modName.Contains("Traffic Manager")))
+                    {
+                        Log.Info($"Local TM:PE detected: '{modName}' in '{mod.modPath}'");
+                        string folder = mod.modPath.Split(Path.DirectorySeparatorChar).Last();
```

Comment:
Use `Path.GetDirectoryName` instead.

Reply:
I tried that first, but it was struggling because the _filename_ isn't in those mod paths, so it ended up trimming off the parent folder rather than the folder name.

See commit: https://github.com/krzychu124/Cities-Skylines-Traffic-Manager-President-Edition/pull/333/commits/b914f9045167813739f43d4cdbcedfcbfa1d3933

Reply:
You don't need `Split`. 
```
string folder = Path.GetDirectoryName(mod.modPath);
```

Reply:
This works great!